communication/state_machine.cpp

```diff
@@ -189,9 +189,9 @@ bool State_machine::update(State_machine* state_machine)
         mode_new ^= MAV_MODE_FLAG_GUIDED_ENABLED; // toggle guided flag
     }
 
-    if(!state_machine->set_mode_stabilize(mav_modes_is_stabilise(mode_new)))
+    if(!state_machine->set_mode_stabilize(mav_modes_is_stabilize(mode_new)))
     {
-        mode_new ^= MAV_MODE_FLAG_STABILIZE_ENABLED; // toggle stabilize flag   
+        mode_new ^= MAV_MODE_FLAG_STABILIZE_ENABLED; // toggle stabilize flag
```

Comment:
Wouldn't be the &= ~ MAV_MODE_... more safe? 
Even if we do check if the flag is set in the function mav_modes_is_stabilize, we would be sure never to toggle the flag involuntary.